ggshield/verticals/secret/secret_scan_collection.py

```diff
@@ -129,18 +160,7 @@ def __init__(
         self.optional_header = optional_header
         self.extra_info = extra_info
 
-        (
-            self.known_secrets_count,
-            self.new_secrets_count,
-        ) = self._get_known_new_secrets_count()
-
-    @property
-    def has_new_secrets(self) -> bool:
-        return self.new_secrets_count > 0
-
-    @property
-    def has_secrets(self) -> bool:
-        return (self.new_secrets_count + self.known_secrets_count) > 0
+        self.all_results = list(self.get_all_results())
```

Comment:
We can do without - I feel like the interface of SecretScanCollection is kinda flawed no matter what we do though, because of how it's mixing up results, nested scan results, etc.

Changed to `total_policy_breaks_count` -> more correct, as Results can now be empty of policy breaks